.github/workflows/E2E.yaml

```
@@ -79,6 +79,17 @@ jobs:
           Add-Content -Encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "githubOwner=$githubOwner"
           Write-Host "githubOwner=$githubOwner"
 
+  CheckRef:
+    runs-on: [ ubuntu-latest ]
+    needs: [ Check ]
+    if: github.event.inputs.ref == ''
```

```suggestion
    if: (github.event.inputs.ref == '') || (github.event.inputs.ref == github.sha)
```

I myself sometimes provide the latest SHA as an argument, even though it doesn't matter.